Tests/AblyChatTests/RoomLifecycleManagerTests.swift

```diff
@@ -125,6 +125,54 @@ struct RoomLifecycleManagerTests {
         }
     }
 
+    // @spec CHA-RL1d
+    @Test
+    func attach_ifOtherOperationInProgress_waitsForItToComplete() async throws {
+        // Given: A RoomLifecycleManager with a DETACH lifecycle operation in progress (the fact that it is a DETACH is not important; it is just an operation whose execution it is easy to prolong and subsequently complete, which is helpful for this test)
+        let contributorDetachOperation = SignallableChannelOperation()
+        let manager = await createManager(
+            contributors: [
+                createContributor(
+                    // Arbitrary, allows the ATTACH to eventually complete
+                    attachBehavior: .complete(.success),
+                    // This allows us to prolong the execution of the DETACH triggered in (1)
+                    detachBehavior: contributorDetachOperation.behavior
+                ),
+            ]
+        )
+
+        let detachOperationID = UUID()
+        let attachOperationID = UUID()
+
+        let statusChangeSubscription = await manager.onChange(bufferingPolicy: .unbounded)
+        // Wait for the manager to enter DETACHING; this our sign that the DETACH operation triggered in (1) has started
+        async let detachingStatusChange = statusChangeSubscription.first { $0.current == .detaching }
+
+        // (1) This is the "DETACH lifecycle operation in progress" mentioned in Given
+        async let _ = manager.performDetachOperation(testsOnly_forcingOperationID: detachOperationID)
+        _ = await detachingStatusChange
+
+        let operationWaitEventSubscription = await manager.testsOnly_subscribeToOperationWaitEvents()
+        async let attachedWaitingForDetachedEvent = operationWaitEventSubscription.first { operationWaitEvent in
+            operationWaitEvent == .init(waitingOperationID: attachOperationID, waitedOperationID: detachOperationID)
+        }
+
+        // When: `performAttachOperation()` is called on the lifecycle manager
+        async let attachResult: Void = manager.performAttachOperation(testsOnly_forcingOperationID: attachOperationID)
+
+        // Then:
+        // - the manager informs us that the ATTACH operation is waiting for the DETACH operation to complete
+        // - when the DETACH completes, the ATTACH operation proceeds (which we check here by verifying that it eventually completes) — note that (as far as I can tell) there is no way to test that the ATTACH operation would have proceeded _only if_ the DETACH had completed; the best we can do is allow the manager to tell us that that this is indeed what it’s doing (which is what we check for in the previous bullet)
```

Comment:
I'm not sure I understand this... surely we _can_ test this to be the case by just waiting for a set time e.g. 10 secs _and_ not marking the detach operation as completed. In turn, we can check the attach operation does not complete in that time?

Reply:
1. it just tells you that the attach operation didn't complete within 10 seconds; what if it were to complete within 11 seconds, say?
2. now you have a 10-second wait in your test (I know that your 10 seconds is arbitrary, but it's best to not have any waits if possible)

Reply:
> thinking out loud but could you not force attach to complete similar to how you have with contributorDetachOperation.complete(result: .success /* arbitrary */ )

No — the `contributorDetachOperation(…)` that you've mentioned forces a _contributor_ operation to complete, not a _room lifecycle operation_.

Reply:
ahh cool - makes sense